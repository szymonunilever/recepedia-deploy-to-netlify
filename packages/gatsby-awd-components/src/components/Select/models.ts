import { LanguageEntry } from '../CountrySelector';

export interface Option {
  value: string;
  label: string;
}
export interface SelectProps {
  options: Option[] | LanguageEntry[];
  placeholder?: string;
  className?: string;
  searchable?: boolean;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  changeHandler?: (val: any) => void;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  input?: any;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  defaultValue?: any;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  formatOptionLabel?: any;
}
