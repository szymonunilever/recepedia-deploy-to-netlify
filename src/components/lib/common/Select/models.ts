interface Option {
  value: string;
  label: string;
}
export interface SelectProps {
  options: Option[];
  placeholder?: string;
  className?: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  changeHandler?: (val: any) => void;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  input?: any;
}
