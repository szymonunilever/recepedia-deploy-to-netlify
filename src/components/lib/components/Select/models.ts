import { LanguageEntry } from '../CountrySelector/index';

export interface Option {
  value: string;
  label: string;
}
export interface SelectProps {
  options: Option[] | LanguageEntry[];
  placeholder?: string;
  className?: string;
  isSearchable?: boolean;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  changeHandler?: (val: any) => void;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  input?: any;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  defaultValue?: any;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  formatOptionLabel?: any;
}
