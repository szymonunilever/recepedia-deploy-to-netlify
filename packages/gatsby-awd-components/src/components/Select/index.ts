export { default } from './Select';
export { default as Select } from './Select';
export * from './models';
