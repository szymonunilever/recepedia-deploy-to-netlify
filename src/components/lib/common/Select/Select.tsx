import React from 'react';
import { SelectProps } from './models';
import ReactSelect from 'react-select';
import cx from 'classnames';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const Option = (props: any) => {
  const {
    children,
    className,
    isDisabled,
    isFocused,
    isSelected,
    innerRef,
    innerProps,
  } = props;
  return (
    <div
      ref={innerRef}
      className={cx(
        {
          option: true,
          'is-disabled': isDisabled,
          'is-focused': isFocused,
          'is-selected': isSelected,
        },
        className
      )}
      {...innerProps}
    >
      {children}
    </div>
  );
};

const Select = ({
  options,
  className = '',
  placeholder = '',
  input,
}: SelectProps) => {
  const resetStyles = {};
  return (
    <ReactSelect
      {...input}
      styles={resetStyles}
      options={options}
      className={className}
      components={{ Option }}
      placeholder={placeholder}
    />
  );
};

export default Select;
