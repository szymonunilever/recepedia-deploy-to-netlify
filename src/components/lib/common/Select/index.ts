export { default as Select } from './Select';
export * from './models';
