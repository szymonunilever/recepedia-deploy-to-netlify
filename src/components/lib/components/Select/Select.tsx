import React, { useState } from 'react';
import { SelectProps, Option } from './models';
// @ts-ignore
import DropdownSelect from 'react-dropdown-select';

import cx from 'classnames';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const CustomOption = (props: any) => {
  const {
    children,
    className,
    isDisabled,
    isSelected,
    onClick,
    item,
    itemIndex,
  } = props;
  {
    return (
      <div
        role="option"
        key={item.label + itemIndex}
        aria-selected={isSelected}
        onClick={onClick}
        aria-label={item.label}
        className={cx(
          {
            option: true,
            'is-disabled': isDisabled,
            'is-selected': isSelected,
          },
          className
        )}
      >
        {children}
      </div>
    );
  }
};

const Select = ({
  input,
  options,
  className = '',
  placeholder = '',
  changeHandler,
  formatOptionLabel,
  defaultValue,
  searchable = false,
}: SelectProps) => {
  className = cx(className, 'select');
  const [values, setValues] = useState(defaultValue ? [defaultValue] : []);

  const onChange = (values: Option[]) => {
    setValues(values);
    if (changeHandler) {
      changeHandler(values);
    }
  };

  const onItemRenderer = () => {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    return ({ item, itemIndex, props, state, methods }: any) => {
      const Composition = (
        <CustomOption
          itemIndex={itemIndex}
          isDisabled={state.disabled}
          isSelected={!!values.length && values[0].value === item.value}
          onClick={(e: React.MouseEvent<HTMLElement>) => {
            e.preventDefault();
            methods.addItem(item);
          }}
          className={props.className}
          item={item}
        >
          {formatOptionLabel ? formatOptionLabel(item) : item.label}
        </CustomOption>
      );

      return Composition;
    };
  };

  const props = {
    values,
    separator: true,
    options,
    className,
    searchable,
    placeholder,
    onChange,
    itemRenderer: onItemRenderer(),
    closeOnSelect: true,
    ...input,
  };

  return <DropdownSelect {...props} />;
};

export default Select;
