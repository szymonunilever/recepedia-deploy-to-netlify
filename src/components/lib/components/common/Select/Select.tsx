import React from 'react';
import { SelectProps } from './models';
import ReactSelect from 'react-select';
import cx from 'classnames';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const Option = (props: any) => {
  const {
    children,
    className,
    isDisabled,
    isFocused,
    isSelected,
    innerRef,
    innerProps,
  } = props;
  return (
    <div
      ref={innerRef}
      className={cx(
        {
          option: true,
          'is-disabled': isDisabled,
          'is-focused': isFocused,
          'is-selected': isSelected,
        },
        className
      )}
      {...innerProps}
    >
      {children}
    </div>
  );
};

const Select = ({
  options,
  className = '',
  placeholder = '',
  input,
  changeHandler,
}: SelectProps) => {
  className = cx(className, 'select');
  const props = {
    ...input,
    styles: {},
    options,
    className,
    components: { Option },
    placeholder,
  };

  changeHandler ? (props.onChange = changeHandler) : null;

  return <ReactSelect {...props} />;
};

export default Select;
