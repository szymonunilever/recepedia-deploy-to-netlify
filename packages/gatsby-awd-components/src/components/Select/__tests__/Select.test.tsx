import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import Select, { SelectProps } from '../index';
import toJson from 'enzyme-to-json';
import selectOptions from '../../../mocks/select.json';

describe('<Select />', () => {
  let wrapper: ReactWrapper;
  const selectProps: SelectProps = {
    options: selectOptions.options,
    defaultValue: '1',
    input: {
      onChange: jest.fn(),
    },
    changeHandler: val => console.log('val selected', val),
  };

  beforeEach(() => {
    wrapper = mount(<Select {...selectProps} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
