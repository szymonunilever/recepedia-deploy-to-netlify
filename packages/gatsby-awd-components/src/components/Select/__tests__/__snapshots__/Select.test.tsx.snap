// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> matches the snapshot 1`] = `
<Select
  changeHandler={[Function]}
  defaultValue="1"
  input={
    Object {
      "onChange": [MockFunction],
    }
  }
  options={
    Array [
      Object {
        "label": "Question 1",
        "value": "question1",
      },
      Object {
        "label": "Question 2",
        "value": "question2",
      },
      Object {
        "label": "Question 3",
        "value": "question3",
      },
    ]
  }
>
  <t
    addPlaceholder=""
    additionalProps={null}
    autoFocus={false}
    className="select"
    clearOnBlur={true}
    clearOnSelect={true}
    clearable={false}
    closeOnScroll={false}
    closeOnSelect={true}
    color="#0074D9"
    create={false}
    createNewLabel="add {search}"
    debounceDelay={0}
    direction="ltr"
    disabled={false}
    disabledLabel="disabled"
    dropdownGap={5}
    dropdownHandle={true}
    dropdownHeight="300px"
    dropdownPosition="bottom"
    handleKeyDownFn={[Function]}
    itemRenderer={[Function]}
    keepSelectedInList={true}
    labelField="label"
    multi={false}
    name={null}
    noDataLabel="No data"
    onChange={[Function]}
    onClearAll={[Function]}
    onCreateNew={[Function]}
    onDropdownClose={[Function]}
    onDropdownOpen={[Function]}
    onSelectAll={[Function]}
    options={
      Array [
        Object {
          "label": "Question 1",
          "value": "question1",
        },
        Object {
          "label": "Question 2",
          "value": "question2",
        },
        Object {
          "label": "Question 3",
          "value": "question3",
        },
      ]
    }
    placeholder=""
    portal={null}
    searchBy="label"
    searchFn={[Function]}
    searchable={false}
    separator={true}
    sortBy={null}
    valueField="value"
    values={
      Array [
        "1",
      ]
    }
  >
    <t
      onClickOutside={[Function]}
    >
      <div>
        <ReactDropdownSelect
          className="react-dropdown-select select"
          color="#0074D9"
          direction="ltr"
          disabled={false}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex="0"
        >
          <div
            className="react-dropdown-select select css-12zlm52-ReactDropdownSelect e1gzf2xs0"
            color="#0074D9"
            direction="ltr"
            disabled={false}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            tabIndex="0"
          >
            <de
              methods={
                Object {
                  "activeCursorItem": [Function],
                  "addItem": [Function],
                  "areAllSelected": [Function],
                  "clearAll": [Function],
                  "createNew": [Function],
                  "dropDown": [Function],
                  "getInputSize": [Function],
                  "getSelectBounds": [Function],
                  "getSelectRef": [Function],
                  "handleKeyDown": [Function],
                  "isSelected": [Function],
                  "removeItem": [Function],
                  "safeString": [Function],
                  "searchResults": [Function],
                  "selectAll": [Function],
                  "setSearch": [Function],
                  "sortBy": [Function],
                  "toggleSelectAll": [Function],
                }
              }
              props={
                Object {
                  "addPlaceholder": "",
                  "additionalProps": null,
                  "autoFocus": false,
                  "className": "select",
                  "clearOnBlur": true,
                  "clearOnSelect": true,
                  "clearable": false,
                  "closeOnScroll": false,
                  "closeOnSelect": true,
                  "color": "#0074D9",
                  "create": false,
                  "createNewLabel": "add {search}",
                  "debounceDelay": 0,
                  "direction": "ltr",
                  "disabled": false,
                  "disabledLabel": "disabled",
                  "dropdownGap": 5,
                  "dropdownHandle": true,
                  "dropdownHeight": "300px",
                  "dropdownPosition": "bottom",
                  "handleKeyDownFn": [Function],
                  "itemRenderer": [Function],
                  "keepOpen": undefined,
                  "keepSelectedInList": true,
                  "labelField": "label",
                  "multi": false,
                  "name": null,
                  "noDataLabel": "No data",
                  "onChange": [Function],
                  "onClearAll": [Function],
                  "onCreateNew": [Function],
                  "onDropdownClose": [Function],
                  "onDropdownOpen": [Function],
                  "onSelectAll": [Function],
                  "options": Array [
                    Object {
                      "label": "Question 1",
                      "value": "question1",
                    },
                    Object {
                      "label": "Question 2",
                      "value": "question2",
                    },
                    Object {
                      "label": "Question 3",
                      "value": "question3",
                    },
                  ],
                  "placeholder": "",
                  "portal": null,
                  "searchBy": "label",
                  "searchFn": [Function],
                  "searchable": false,
                  "separator": true,
                  "sortBy": null,
                  "valueField": "value",
                  "values": Array [
                    "1",
                  ],
                }
              }
              state={
                Object {
                  "cursor": null,
                  "dropdown": false,
                  "search": "",
                  "selectBounds": Object {
                    "bottom": 0,
                    "height": 0,
                    "left": 0,
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  },
                  "values": Array [
                    "1",
                  ],
                }
              }
            >
              <ContentComponent
                className="react-dropdown-select-content react-dropdown-select-type-single"
                onClick={[Function]}
              >
                <div
                  className="react-dropdown-select-content react-dropdown-select-type-single css-jznujr-ContentComponent e1gn6jc30"
                  onClick={[Function]}
                >
                  <span />
                  <t
                    methods={
                      Object {
                        "activeCursorItem": [Function],
                        "addItem": [Function],
                        "areAllSelected": [Function],
                        "clearAll": [Function],
                        "createNew": [Function],
                        "dropDown": [Function],
                        "getInputSize": [Function],
                        "getSelectBounds": [Function],
                        "getSelectRef": [Function],
                        "handleKeyDown": [Function],
                        "isSelected": [Function],
                        "removeItem": [Function],
                        "safeString": [Function],
                        "searchResults": [Function],
                        "selectAll": [Function],
                        "setSearch": [Function],
                        "sortBy": [Function],
                        "toggleSelectAll": [Function],
                      }
                    }
                    props={
                      Object {
                        "addPlaceholder": "",
                        "additionalProps": null,
                        "autoFocus": false,
                        "className": "select",
                        "clearOnBlur": true,
                        "clearOnSelect": true,
                        "clearable": false,
                        "closeOnScroll": false,
                        "closeOnSelect": true,
                        "color": "#0074D9",
                        "create": false,
                        "createNewLabel": "add {search}",
                        "debounceDelay": 0,
                        "direction": "ltr",
                        "disabled": false,
                        "disabledLabel": "disabled",
                        "dropdownGap": 5,
                        "dropdownHandle": true,
                        "dropdownHeight": "300px",
                        "dropdownPosition": "bottom",
                        "handleKeyDownFn": [Function],
                        "itemRenderer": [Function],
                        "keepOpen": undefined,
                        "keepSelectedInList": true,
                        "labelField": "label",
                        "multi": false,
                        "name": null,
                        "noDataLabel": "No data",
                        "onChange": [Function],
                        "onClearAll": [Function],
                        "onCreateNew": [Function],
                        "onDropdownClose": [Function],
                        "onDropdownOpen": [Function],
                        "onSelectAll": [Function],
                        "options": Array [
                          Object {
                            "label": "Question 1",
                            "value": "question1",
                          },
                          Object {
                            "label": "Question 2",
                            "value": "question2",
                          },
                          Object {
                            "label": "Question 3",
                            "value": "question3",
                          },
                        ],
                        "placeholder": "",
                        "portal": null,
                        "searchBy": "label",
                        "searchFn": [Function],
                        "searchable": false,
                        "separator": true,
                        "sortBy": null,
                        "valueField": "value",
                        "values": Array [
                          "1",
                        ],
                      }
                    }
                    state={
                      Object {
                        "cursor": null,
                        "dropdown": false,
                        "search": "",
                        "selectBounds": Object {
                          "bottom": 0,
                          "height": 0,
                          "left": 0,
                          "right": 0,
                          "top": 0,
                          "width": 0,
                        },
                        "values": Array [
                          "1",
                        ],
                      }
                    }
                  >
                    <InputComponent
                      className="react-dropdown-select-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onKeyPress={[Function]}
                      placeholder=""
                      readOnly={true}
                      size={0}
                      tabIndex="-1"
                      value=""
                    >
                      <input
                        className="react-dropdown-select-input css-26pbll-InputComponent e11wid6y0"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onKeyPress={[Function]}
                        placeholder=""
                        readOnly={true}
                        size={0}
                        tabIndex="-1"
                        value=""
                      />
                    </InputComponent>
                  </t>
                </div>
              </ContentComponent>
            </de>
            <Ae
              methods={
                Object {
                  "activeCursorItem": [Function],
                  "addItem": [Function],
                  "areAllSelected": [Function],
                  "clearAll": [Function],
                  "createNew": [Function],
                  "dropDown": [Function],
                  "getInputSize": [Function],
                  "getSelectBounds": [Function],
                  "getSelectRef": [Function],
                  "handleKeyDown": [Function],
                  "isSelected": [Function],
                  "removeItem": [Function],
                  "safeString": [Function],
                  "searchResults": [Function],
                  "selectAll": [Function],
                  "setSearch": [Function],
                  "sortBy": [Function],
                  "toggleSelectAll": [Function],
                }
              }
              props={
                Object {
                  "addPlaceholder": "",
                  "additionalProps": null,
                  "autoFocus": false,
                  "className": "select",
                  "clearOnBlur": true,
                  "clearOnSelect": true,
                  "clearable": false,
                  "closeOnScroll": false,
                  "closeOnSelect": true,
                  "color": "#0074D9",
                  "create": false,
                  "createNewLabel": "add {search}",
                  "debounceDelay": 0,
                  "direction": "ltr",
                  "disabled": false,
                  "disabledLabel": "disabled",
                  "dropdownGap": 5,
                  "dropdownHandle": true,
                  "dropdownHeight": "300px",
                  "dropdownPosition": "bottom",
                  "handleKeyDownFn": [Function],
                  "itemRenderer": [Function],
                  "keepOpen": undefined,
                  "keepSelectedInList": true,
                  "labelField": "label",
                  "multi": false,
                  "name": null,
                  "noDataLabel": "No data",
                  "onChange": [Function],
                  "onClearAll": [Function],
                  "onCreateNew": [Function],
                  "onDropdownClose": [Function],
                  "onDropdownOpen": [Function],
                  "onSelectAll": [Function],
                  "options": Array [
                    Object {
                      "label": "Question 1",
                      "value": "question1",
                    },
                    Object {
                      "label": "Question 2",
                      "value": "question2",
                    },
                    Object {
                      "label": "Question 3",
                      "value": "question3",
                    },
                  ],
                  "placeholder": "",
                  "portal": null,
                  "searchBy": "label",
                  "searchFn": [Function],
                  "searchable": false,
                  "separator": true,
                  "sortBy": null,
                  "valueField": "value",
                  "values": Array [
                    "1",
                  ],
                }
              }
              state={
                Object {
                  "cursor": null,
                  "dropdown": false,
                  "search": "",
                  "selectBounds": Object {
                    "bottom": 0,
                    "height": 0,
                    "left": 0,
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  },
                  "values": Array [
                    "1",
                  ],
                }
              }
            >
              <SeparatorComponent
                className="react-dropdown-select-separator"
              >
                <div
                  className="react-dropdown-select-separator css-vxbae9-SeparatorComponent e19h5j1v0"
                />
              </SeparatorComponent>
            </Ae>
            <Ee
              methods={
                Object {
                  "activeCursorItem": [Function],
                  "addItem": [Function],
                  "areAllSelected": [Function],
                  "clearAll": [Function],
                  "createNew": [Function],
                  "dropDown": [Function],
                  "getInputSize": [Function],
                  "getSelectBounds": [Function],
                  "getSelectRef": [Function],
                  "handleKeyDown": [Function],
                  "isSelected": [Function],
                  "removeItem": [Function],
                  "safeString": [Function],
                  "searchResults": [Function],
                  "selectAll": [Function],
                  "setSearch": [Function],
                  "sortBy": [Function],
                  "toggleSelectAll": [Function],
                }
              }
              onClick={[Function]}
              props={
                Object {
                  "addPlaceholder": "",
                  "additionalProps": null,
                  "autoFocus": false,
                  "className": "select",
                  "clearOnBlur": true,
                  "clearOnSelect": true,
                  "clearable": false,
                  "closeOnScroll": false,
                  "closeOnSelect": true,
                  "color": "#0074D9",
                  "create": false,
                  "createNewLabel": "add {search}",
                  "debounceDelay": 0,
                  "direction": "ltr",
                  "disabled": false,
                  "disabledLabel": "disabled",
                  "dropdownGap": 5,
                  "dropdownHandle": true,
                  "dropdownHeight": "300px",
                  "dropdownPosition": "bottom",
                  "handleKeyDownFn": [Function],
                  "itemRenderer": [Function],
                  "keepOpen": undefined,
                  "keepSelectedInList": true,
                  "labelField": "label",
                  "multi": false,
                  "name": null,
                  "noDataLabel": "No data",
                  "onChange": [Function],
                  "onClearAll": [Function],
                  "onCreateNew": [Function],
                  "onDropdownClose": [Function],
                  "onDropdownOpen": [Function],
                  "onSelectAll": [Function],
                  "options": Array [
                    Object {
                      "label": "Question 1",
                      "value": "question1",
                    },
                    Object {
                      "label": "Question 2",
                      "value": "question2",
                    },
                    Object {
                      "label": "Question 3",
                      "value": "question3",
                    },
                  ],
                  "placeholder": "",
                  "portal": null,
                  "searchBy": "label",
                  "searchFn": [Function],
                  "searchable": false,
                  "separator": true,
                  "sortBy": null,
                  "valueField": "value",
                  "values": Array [
                    "1",
                  ],
                }
              }
              state={
                Object {
                  "cursor": null,
                  "dropdown": false,
                  "search": "",
                  "selectBounds": Object {
                    "bottom": 0,
                    "height": 0,
                    "left": 0,
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  },
                  "values": Array [
                    "1",
                  ],
                }
              }
            >
              <DropdownHandleComponent
                className="react-dropdown-select-dropdown-handle"
                color="#0074D9"
                dropdownOpen={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                tabIndex="-1"
              >
                <div
                  className="react-dropdown-select-dropdown-handle css-6wsm8k-DropdownHandleComponent e1vudypg0"
                  color="#0074D9"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  tabIndex="-1"
                >
                  <svg
                    fill="currentColor"
                    viewBox="0 0 40 40"
                  >
                    <path
                      d="M31 26.4q0 .3-.2.5l-1.1 1.2q-.3.2-.6.2t-.5-.2l-8.7-8.8-8.8 8.8q-.2.2-.5.2t-.5-.2l-1.2-1.2q-.2-.2-.2-.5t.2-.5l10.4-10.4q.3-.2.6-.2t.5.2l10.4 10.4q.2.2.2.5z"
                    />
                  </svg>
                </div>
              </DropdownHandleComponent>
            </Ee>
          </div>
        </ReactDropdownSelect>
      </div>
    </t>
  </t>
</Select>
`;
